### PR TITLE
Add remaining screens

### PR DESCRIPTION
**Pull request by cherrybeans**
- Simplify logging in UserAuth and add method to check if user is logged in
- Make loading screen automatically load the correct menu screen when done
- Add Lobby screen, Join lobby screen, Highscore screen, Tutorial screen, Gameplay screen (improved visuals)
- Complete navigation structure between screens
- Clean up code for consistency
- Connect login form to firebase auth

Screenshot_20210319_201029

Screenshot_20210319_201042

Problems: 
- When pressing a button, the event renders many times, e.g. when clicking log out button (TextButton with onClick event), the function is called like over 10 times. When creating a button that signs you in, Fireauth malfunctions because the event is sent so many times. Created a separate issue for fixing this #29 
- We are duplicating a lot of styles. Created a separate issue for looking into refactoring this #28

### CODE REVIEW

File: core/src/main/kotlin/no/ntnu/beardblaster/user/UserAuth.kt [hunk 1]
```diff
@@ -1,54 +1,64 @@
...
         GdxFIRAuth.inst()
                 .signInWithEmailAndPassword(email, password.toCharArray())
                 .then<GdxFirebaseUser> { gdxFirebaseUser ->
-                    Gdx.app.debug("Sign in user", "Signed in. currentUser: " + GdxFIRAuth.inst().currentUser?.userInfo?.email)
+                    LOG.debug { "Signed in. currentUser: ${GdxFIRAuth.inst().currentUser?.userInfo?.email}" }
                 }
                 .fail { s, _ ->
                     if (s.contains("")) {
-                        Gdx.app.debug("Sign in user", "No user exists with those credentials" + s)
+                        LOG.debug { "No user exists with those credentials: ($email, $password) - $s" }
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/user/UserAuth.kt]: No logging of the users password attempts in clear-text! 🗡️  

Suggestion: 
```java 
LOG.debug { "Login failed: $s" }
```

File: core/src/main/kotlin/no/ntnu/beardblaster/user/UserAuth.kt [hunk 2]
```diff
@@ -1,54 +1,64 @@
...
 
     override fun signIn(email: String, password: String) {
         GdxFIRAuth.inst()
                 .signInWithEmailAndPassword(email, password.toCharArray())
                 .then<GdxFirebaseUser> { gdxFirebaseUser ->
-                    Gdx.app.debug("Sign in user", "Signed in. currentUser: " + GdxFIRAuth.inst().currentUser?.userInfo?.email)
+                    LOG.debug { "Signed in. currentUser: ${GdxFIRAuth.inst().currentUser?.userInfo?.email}" }
                 }
                 .fail { s, _ ->
                     if (s.contains("")) {
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/user/UserAuth.kt]: I know this is some legacy code right now, but maybe just remove this `if`-statement?

File: core/src/main/kotlin/no/ntnu/beardblaster/user/UserAuth.kt [hunk 3]
```diff
@@ -1,54 +1,64 @@
...
     }
 
     override fun signOut() {
         GdxFIRAuth.inst().signOut()
                 .then<Void> {
-                    Gdx.app.debug("Sign out user", "Signed out. currentUser: " + GdxFIRAuth.inst().currentUser)
+                    LOG.debug { "Signed out. currentUser: ${GdxFIRAuth.inst().currentUser}" }
                 }
                 .fail { s, _ ->
                     if (s.contains("")) {
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/user/UserAuth.kt]: Same as above comment, maybe this `if`-statement could just be removed (it checks if a String message contains an empty string).

**Review by gullskatten — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/no/ntnu/beardblaster/BeardBlasterGame.kt [hunk 4]
```diff
@@ -31,15 +32,24 @@ class BeardBlasterGame : KtxGame<AbstractScreen>() {
         //Set debug level
         Gdx.app.logLevel = Application.LOG_DEBUG
         LOG.debug { "Create game instance" }
+        // Speed login
+        // if (!UserAuth().isLoggedIn()) {
+        //    UserAuth().signIn("beard@blaster.com", "beardblaster")
+        //}
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/BeardBlasterGame.kt]: Remove this commented out code? 🌵

File: core/src/main/kotlin/no/ntnu/beardblaster/view/JoinLobbyScreen.kt [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        table.setBounds(0f, 0f, viewport.worldWidth, viewport.worldHeight)
+
+        // Fonts
+        val standardFont = Assets.assetManager.get(Assets.standardFont)
+
+        // Creating buttons
+        val textButtonStyle = TextButton.TextButtonStyle()
+        skin.getDrawable("button_default").also { textButtonStyle.up = it }
+        skin.getDrawable("button_default_hover").also { textButtonStyle.over = it }
+        skin.getDrawable("button_default_pressed").also { textButtonStyle.down = it }
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/view/JoinLobbyScreen.kt]: As a comment, we should really soon plan on setting up either factory-methods for creating these buttons with slightly different styles (e.g., colors: green, red, purple, orange) (we've duplicated the skin / styling so many times already) 😿

File: core/src/main/kotlin/no/ntnu/beardblaster/view/LoadingScreen.kt [hunk 6]
```diff
@@ -30,34 +32,34 @@ class LoadingScreen(game: BeardBlasterGame) : AbstractScreen(game) {
...
         progress = Assets.assetManager.progress.roundToInt().toFloat()
-        //Check iff assetManager is done loading
+        // Check iff assetManager is done loading
         if (Assets.assetManager.update()) {
-            if (Gdx.input.justTouched()) {
-                //Change screen
-                game.setScreen<LoginMenuScreenScreen>()
-            }
+            // Change screen
+            if (UserAuth().isLoggedIn()) {
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/view/LoadingScreen.kt:37]: 🙌 Nice!

File: core/src/main/kotlin/no/ntnu/beardblaster/view/JoinLobbyScreen.kt (hunk 5 above)
Comment (cherrybeans) [reply core/src/main/kotlin/no/ntnu/beardblaster/view/JoinLobbyScreen.kt]: Agreed! I created an issue for it in #28.

File: core/src/main/kotlin/no/ntnu/beardblaster/view/LoginScreen.kt [hunk 7]
```diff
@@ -119,13 +118,17 @@ class LoginScreen(game: BeardBlasterGame) : AbstractScreen(game) {
...
         loginStage.addActor(table)
 
     }
 
     override fun update(delta: Float) {
         loginButton.onClick {
+            if (!UserAuth().isLoggedIn() && emailInput.text.isNotEmpty() && passwordInput.text.isNotEmpty()) {
+                UserAuth().signIn(emailInput.text, passwordInput.text)
+            }
             game.setScreen<MenuScreen>()
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/view/LoginScreen.kt:132]: Vi kan rydde opp i dette senere, men nå kommer man uansett til MenuScreen selv om username / password-feltet er tomt eller brukeren ikke eksisterer.

**Review by gullskatten — APPROVED** [review]

File: core/src/main/kotlin/no/ntnu/beardblaster/view/LoginScreen.kt (hunk 7 above)
Comment (cherrybeans) [reply core/src/main/kotlin/no/ntnu/beardblaster/view/LoginScreen.kt:132]: Jepp, tenkte bare å ikke sende mange unødvendige requests til Firebase under utvikling

**Pull request merged**